test failure: test_run_celery fails with JSONDecodeError

See https://github.com/iterative/dvc/actions/runs/3475811796/jobs/5810969556#step:5:75. This looks like a real bug.

cc @pmrowla @karajan1001 

```python
=================================== FAILURES ===================================
_______________________________ test_run_celery ________________________________
[gw2] darwin -- Python 3.9.14 /Users/runner/hostedtoolcache/Python/3.9.14/x64/bin/python

tmp_dir = PosixTmpDir('/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/popen-gw2/test_run_celery0')
scm = Git: '/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/popen-gw2/test_run_celery0/.git'
dvc = Repo: '/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pytest-of-runner/pytest-0/popen-gw2/test_run_celery0'
exp_stage = Stage: 'copy-file'
mocker = <pytest_mock.plugin.MockerFixture object at 0x11b2b0490>

    def test_run_celery(tmp_dir, scm, dvc, exp_stage, mocker):
        """Test running with full (non-pytest-celery) dvc-task queue."""
        dvc.experiments.run(exp_stage.addressing, params=["foo=2"], queue=True)
        dvc.experiments.run(exp_stage.addressing, params=["foo=3"], queue=True)
        assert len(dvc.experiments.stash_revs) == 2
    
        repro_spy = mocker.spy(dvc.experiments, "reproduce_celery")
>       results = dvc.experiments.run(run_all=True)

/Users/runner/work/dvc/dvc/tests/func/experiments/test_experiments.py:537: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/runner/work/dvc/dvc/dvc/repo/experiments/__init__.py:488: in run
    return run(self.repo, *args, **kwargs)
/Users/runner/work/dvc/dvc/dvc/repo/__init__.py:48: in wrapper
    return f(repo, *args, **kwargs)
/Users/runner/work/dvc/dvc/dvc/repo/experiments/run.py:32: in run
    return repo.experiments.reproduce_celery(jobs=jobs)
<string>:3: in reproduce_celery
    ???
/Users/runner/hostedtoolcache/Python/3.9.14/x[64](https://github.com/iterative/dvc/actions/runs/3475811796/jobs/5810969556#step:5:65)/lib/python3.9/unittest/mock.py:1092: in __call__
    return self._mock_call(*args, **kwargs)
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/unittest/mock.py:1096: in _mock_call
    return self._execute_mock_call(*args, **kwargs)
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/unittest/mock.py:1157: in _execute_mock_call
    result = effect(*args, **kwargs)
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/site-packages/pytest_mock/plugin.py:138: in wrapper
    r = method(*args, **kwargs)
/Users/runner/work/dvc/dvc/dvc/repo/experiments/__init__.py:199: in reproduce_celery
    while not self.celery_queue.proc.get(entry.stash_rev):
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/site-packages/dvc_task/proc/manager.py:[69](https://github.com/iterative/dvc/actions/runs/3475811796/jobs/5810969556#step:5:70): in get
    return self[key]
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/site-packages/dvc_task/proc/manager.py:51: in __getitem__
    return ProcessInfo.load(info_path)
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/site-packages/dvc_task/proc/process.py:43: in load
    return cls.from_dict(json.load(fobj))
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/json/__init__.py:293: in load
    return loads(fp.read(),
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/json/__init__.py:346: in loads
    return _default_decoder.decode(s)
/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.decoder.JSONDecoder object at 0x10b4b[76](https://github.com/iterative/dvc/actions/runs/3475811796/jobs/5810969556#step:5:77)d0>, s = '', idx = 0

    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

/Users/runner/hostedtoolcache/Python/3.9.14/x64/lib/python3.9/json/decoder.py:355: JSONDecodeError
```

A different one, see https://github.com/iterative/dvc/actions/runs/3510730975/jobs/5880814259.

```python
FAILED tests/func/experiments/test_show.py::test_show_failed_experiment - PermissionError: [Errno 13] Permission denied: '\\\\?\\C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw1\\test_show_failed_experiment0\\.dvc\\tmp\\exps\\celery\\broker\\in\\1412500_45926eca-ea98-40dc-92d6-6c35bb159aac.dvc-exp-51e250-1@localhost.celery.pidbox.msg'

```

```python
FAILED tests/func/experiments/test_experiments.py::test_apply_failed - dulwich.file.FileLocked: (b'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw0\\test_apply_failed0\\.git\\refs\\exps\\celery\\stash', b'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw0\\test_apply_failed0\\.git\\refs\\exps\\celery\\stash.lock')
```

```python
================================== FAILURES ===================================
______________________________ test_apply_failed ______________________________
[gw0] win32 -- Python 3.9.13 C:\hostedtoolcache\windows\Python\3.9.13\x64\python.exe

self = <dulwich.file._GitFile object at 0x0000020580E58880>
filename = b'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw0\\test_apply_failed0\\.git\\refs\\exps\\celery\\stash'
mode = 'wb', bufsize = -1, mask = 420

    def __init__(self, filename, mode, bufsize, mask):
        self._filename = filename
        if isinstance(self._filename, bytes):
            self._lockfilename = self._filename + b".lock"
        else:
            self._lockfilename = self._filename + ".lock"
        try:
>           fd = os.open(
                self._lockfilename,
                os.O_RDWR | os.O_CREAT | os.O_EXCL | getattr(os, "O_BINARY", 0),
                mask,
            )
E           FileExistsError: [Errno 17] File exists: b'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw0\\test_apply_failed0\\.git\\refs\\exps\\celery\\stash.lock'

C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\dulwich\file.py:150: FileExistsError

The above exception was the direct cause of the following exception:

tmp_dir = WindowsTmpDir('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/popen-gw0/test_apply_failed0')
scm = Git: 'C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\popen-gw0\test_apply_failed0\.git'
dvc = Repo: 'C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\popen-gw0\test_apply_failed0'
failed_exp_stage = Stage: 'failed-copy-file'
test_queue = <dvc.repo.experiments.queue.celery.LocalCeleryQueue object at 0x00000205FAB2EDC0>

    def test_apply_failed(tmp_dir, scm, dvc, failed_exp_stage, test_queue):
        from dvc.exceptions import InvalidArgumentError
    
        dvc.experiments.run(
            failed_exp_stage.addressing, params=["foo=2"], queue=True
        )
        exp_rev = dvc.experiments.scm.resolve_rev(f"{CELERY_STASH}@{{0}}")
    
>       dvc.experiments.run(
            failed_exp_stage.addressing, params=["foo=3"], queue=True, name="foo"
        )

D:\a\dvc\dvc\tests\func\experiments\test_experiments.py:148: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
D:\a\dvc\dvc\dvc\repo\experiments\__init__.py:469: in run
    return run(self.repo, *args, **kwargs)
D:\a\dvc\dvc\dvc\repo\__init__.py:48: in wrapper
    return f(repo, *args, **kwargs)
D:\a\dvc\dvc\dvc\repo\experiments\run.py:78: in run
    queue_entry = repo.experiments.queue_one(
D:\a\dvc\dvc\dvc\repo\experiments\__init__.py:149: in queue_one
    return self.new(
D:\a\dvc\dvc\dvc\repo\experiments\__init__.py:[26](https://github.com/iterative/dvc/actions/runs/3854804740/jobs/6569190255#step:5:27)4: in new
    return queue.put(*args, **kwargs)
D:\a\dvc\dvc\dvc\repo\experiments\queue\celery.py:177: in put
    entry = self._stash_exp(*args, **kwargs)
D:\a\dvc\dvc\dvc\repo\experiments\queue\base.py:4[27](https://github.com/iterative/dvc/actions/runs/3854804740/jobs/6569190255#step:5:28): in _stash_exp
    stash_rev = self.stash.push(message=msg)
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\scmrepo\git\stash.py:40: in push
    rev, reset = self.scm._stash_push(  # pylint: disable=protected-access
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\scmrepo\git\__init__.py:[28](https://github.com/iterative/dvc/actions/runs/3854804740/jobs/6569190255#step:5:29)9: in _backend_func
    result = func(*args, **kwargs)
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\scmrepo\git\backend\dulwich\__init__.py:736: in _stash_push
    rev = stash.push(message=message_b)
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\dulwich\stash.py:122: in push
    cid = self._repo.do_commit(
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\dulwich\repo.py:1020: in do_commit
    ok = self.refs.set_if_equals(
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\dulwich\refs.py:853: in set_if_equals
    with GitFile(filename, "wb") as f:
C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\dulwich\file.py:92: in GitFile
    return _GitFile(filename, mode, bufsize, mask)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <dulwich.file._GitFile object at 0x0000020580E58880>
filename = b'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw0\\test_apply_failed0\\.git\\refs\\exps\\celery\\stash'
mode = 'wb', bufsize = -1, mask = 420

    def __init__(self, filename, mode, bufsize, mask):
        self._filename = filename
        if isinstance(self._filename, bytes):
            self._lockfilename = self._filename + b".lock"
        else:
            self._lockfilename = self._filename + ".lock"
        try:
            fd = os.open(
                self._lockfilename,
                os.O_RDWR | os.O_CREAT | os.O_EXCL | getattr(os, "O_BINARY", 0),
                mask,
            )
        except FileExistsError as exc:
>           raise FileLocked(filename, self._lockfilename) from exc
E           dulwich.file.FileLocked: (b'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw0\\test_apply_failed0\\.git\\refs\\exps\\celery\\stash', b'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\popen-gw0\\test_apply_failed0\\.git\\refs\\exps\\celery\\stash.lock')

C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\dulwich\file.py:1[56](https://github.com/iterative/dvc/actions/runs/3854804740/jobs/6569190255#step:5:57): FileLocked
```

See https://github.com/iterative/dvc/actions/runs/3854804740/jobs/6569190255#step:5:27.

Another failure in `test_apply_failed`:
```python
            if not ok:
                # Fail if the atomic compare-and-swap failed, leaving the
                # commit and all its objects as garbage.
>               raise CommitError(f"{ref!r} changed during commit")
E               dulwich.errors.CommitError: b'refs/exps/celery/stash' changed during commit

C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\dulwich\repo.py:1045: CommitError
```

See https://github.com/iterative/dvc/actions/runs/3854804740/jobs/6569190033#step:5:27.

Another failure on `test_apply_failed`:
https://github.com/iterative/dvc/actions/runs/3981910162/jobs/6825952875#step:6:61
```
FAILED tests/func/experiments/test_experiments.py::test_apply_failed - dvc.repo.experiments.exceptions.InvalidExpRevError: '50bbdf1b310c9ebd95374b502d6fe1dcf6d679f6' does not appear to be an experiment commit.
```

--- a/tests/func/experiments/test_apply.py
+++ b/tests/func/experiments/test_apply.py
@@ -1,4 +1,3 @@
-import dulwich.errors
 import pytest
 from funcy import first
 
@@ -27,14 +26,6 @@ def test_apply(tmp_dir, scm, dvc, exp_stage):
     assert (tmp_dir / "metrics.yaml").read_text().strip() == "foo: 3"
 
 
-@pytest.mark.xfail(
-    raises=(
-        dulwich.errors.CommitError,
-        dulwich.file.FileLocked,
-    ),
-    strict=False,
-    reason="See https://github.com/iterative/dvc/issues/8570",
-)
 def test_apply_failed(tmp_dir, scm, dvc, failed_exp_stage, test_queue):
     from dvc.exceptions import InvalidArgumentError
 

